Add docs for Reaper Control Plane mode (#1398)
k8ssandra/k8ssandra-operator: docs/content/en/components/reaper/_index.md

In Cassandra 4.0, internal system “diagnostic events” have become available, via the work done in CASSANDRA-12944. These allow operators to observe internal Cassandra events, for example in unit tests and with external tools. These diagnostic events provide operational monitoring and troubleshooting beyond logs and metrics.

> Reaper does not need an external database to store its data. It can run with a _local_ storage, which means Reaper will persist its in-memory data to the local file system. Reaper can also use Postgres and Cassandra itself as a storage backend for its data, and is capable of repairing all Cassandra versions since 1.2 up to the latest 4.0.

Reviewer: **Todo:** We cannot use Postgres anymore so we should remove mentions of it.

    deploymentMode: SINGLE
```

### Deploying a control plane Reaper

In order to deploy a control plane Reaper, you'll need to create the Reaper object directly:

```yaml
kind: Reaper
metadata:
  name: cp-reaper
spec:
  storageType: local
  storageConfig:
    accessModes:
      - ReadWriteOnce
    resources:
      requests:
        storage: 256Mi
  httpManagement:
    enabled: true
```

The _control plane_ Reaper must use a _local_ storage because there's no Cassandra cluster yet. 

> The local storage requires its own config, mostly to ensure enough disk space is allocated to Reaper. You can skip the `storageType` field, which will make the k8s cluster use whatever the default storage class is.

Reviewer: ```suggestion
The local storage requires its own config, mostly to ensure enough disk space is allocated to Reaper. You can skip the `storageClassName` field to use the default storage class, or set it to specify one explicitly.
```

> It's also a good idea to enable the `httpManagement`. This will make Reaper use HTTP instead of JMX when interacting with Cassandra.

Reviewer: ```suggestion
We recommend enabling the `httpManagement` feature. This will make Reaper use HTTP instead of JMX when interacting with Cassandra.
```

> In order to enrol a k8ssandra cluster in the control plane Reaper, you'll need to add a `reaperRef` to the `reaper` section of the `K8ssandraCluster` manifest:

Reviewer: ```suggestion
In order to enroll a k8ssandra cluster in the control plane Reaper, you'll need to add a `reaperRef` to the `reaper` section of the `K8ssandraCluster` manifest:
```

```yaml
spec:
  reaper:
    reaperRef:

>       name: cp-reaper

Reviewer: **Todo:** it's likely that the reaper object will be in another namespace, right? Could you add the namespace to this sample to make this more obvious? Thanks
